docs/depfile.md

```diff
@@ -36,6 +39,17 @@ analyzer:
   count_use_statements: false
 ```
 
+### `types`
+
+Deptrac can have different parts of the php file as a source for the dependency. By default, only class definitions and use statements can be the source of the dependency and superglobal variable usage is not tracked. To analyze file more fully, you can define what types of `DependencyEmmiters` you want to apply on the analyzed file:
+
+- `class` (default) - analyzes class definitions for everything apart from superglobal usage.
+- `class_superglobal` - analyzes class definitions for superglobal usage.
+- `use` (default) - analyzes file definitions for use statements.
+- `file` - analyzes file for everything apart from use statements and function/class definitions.
+- `function` - analyzes function definitions for everything apart from superglobal usage.
+- `function_superglobal` - analyzes function definitions for superglobal usage.
+
```

Comment:
I am wondering about this paragraph.. doesn't it mean is only possible to scan for superglobals within functions or classes, but not in plain php files?

Reply:
Well, no. See https://github.com/qossmic/deptrac/blob/main/tests/DependencyEmitter/FileDependencyEmitterTest.php#L37. I recall that I differentiated `class_superglobal` as not to break current behavior.

`function_superglobal` - I am not that sure about it. I think `function` support was implemented in a previous commit, so the same reasoning? I don't recall honestly.